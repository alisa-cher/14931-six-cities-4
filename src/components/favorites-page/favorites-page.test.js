import React from 'react';
import renderer from 'react-test-renderer';
import FavoritesPage from "./favorites-page.jsx";
import {BrowserRouter} from "react-router-dom";

const offers = [{
  id: 15,
  bedrooms: 3,
  description: `An independent House`,
  maxAdults: 4,
  type: `appartement`,
  price: 140,
  title: `just amazing1`,
  isPremium: true,
  isFavorite: true,
  rating: 4.3,
  photos: [`img/room.jpg`, `img/studio-01.jpg`],
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: {
    avatar: `img/avatar.svg`,
    isPro: true,
    name: `Angelina`
  },
  previewPhoto: `img/apartment-01.jpg`,
}];

const favoriteOffers = [{
  city: `Amsterdam`,
  offers
}];

it(`Favorites page renders correctly`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <FavoritesPage
            offers={favoriteOffers}
            isUserLoggedIn={true}
            userPhoto={`/img/user-photo.jpg`}
            userEmail={`some-valid-email`}
            onFavoriteButtonClick={() => {}}
            onCardTitleClick={() => {}}
          />
        </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
