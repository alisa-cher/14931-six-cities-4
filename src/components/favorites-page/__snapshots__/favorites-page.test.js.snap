// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites page renders correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="../img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="user avatar"
                    height="41"
                    src="/img/user-photo.jpg"
                    style={
                      Object {
                        "borderRadius": "10px",
                      }
                    }
                    width="81"
                  />
                </div>
                <span
                  className="header__user-name user__name"
                >
                  some-valid-email
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
        style={
          Object {
            "minHeight": "710px",
          }
        }
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Amsterdam
                  </span>
                </a>
              </div>
            </div>
            <div
              className="places__list favorites__list"
            >
              <article
                className="place-card favorites__card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/15"
                    onClick={[Function]}
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      height="200"
                      src="img/apartment-01.jpg"
                      width="260"
                    />
                  </a>
                </div>
                <div
                  className="place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        140
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="button place-card__bookmark-button place-card__bookmark-button--active"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="button place-card__bookmark-icon"
                        height={19}
                        width={18}
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__rating place-card"
                    >
                      <div
                        className="place-card__stars rating__stars"
                      >
                        <span
                          style={
                            Object {
                              "width": "80%",
                            }
                          }
                        />
                        <span
                          className="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                    onClick={[Function]}
                  >
                    <a
                      href="/offer/15"
                      onClick={[Function]}
                    >
                      just amazing1
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Appartement
                  </p>
                </div>
              </article>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="../img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
